dask/highlevelgraph.py

```diff
@@ -24,6 +27,109 @@ def _find_layer_containing_key(key):
     return ret
 
 
+class Layer(collections.abc.Mapping):
+    """High level graph layer
+
+    This abstract class establish a protocol for high level graph layers.
+    """
+
+    def cull(
+        self, keys: Set, all_hlg_keys: Container
```

Comment:
I wonder if `Iterable` is what you're looking for here

Reply:
Fixed